contracts/peripheral/AuraViewHelpers.sol

```diff
@@ -0,0 +1,441 @@
+            periodFinish: rewardPool.periodFinish(),
+            lastUpdateTime: rewardPool.lastUpdateTime(),
+            rewardPerTokenStored: rewardPool.rewardPerTokenStored(),
+            queuedRewards: rewardPool.queuedRewards()
+        });
+
+        return
+            Pool({
+                pid: _pid,
+                lptoken: poolInfo.lptoken,
+                token: poolInfo.token,
+                gauge: poolInfo.gauge,
+                crvRewards: poolInfo.crvRewards,
+                stash: poolInfo.stash,
+                shutdown: poolInfo.shutdown,
+                rewardToken: rewardPool.rewardToken(),
+                poolId: poolId,
+                normalizedWeights: normalizedWeights,
+                poolTokens: poolTokens,
+                underlying: underlying,
+                rewardsData: rewardsData,
+                extraRewards: extraRewards,
+                totalSupply: rewardPool.totalSupply()
+            });
+    }
+
+    function getPoolsBalances(address _booster, address _account) external view returns (PoolBalances[] memory) {
+        uint256 poolLength = IBooster(_booster).poolLength();
+        PoolBalances[] memory balances = new PoolBalances[](poolLength);
+        for (uint256 i = 0; i < poolLength; i++) {
+            IBooster.PoolInfo memory poolInfo = IBooster(_booster).poolInfo(i);
+            balances[i] = getPoolBalances(poolInfo.crvRewards, i, _account);
+        }
+        return balances;
+    }
+
+    function getPoolBalances(
+        address _rewardPool,
+        uint256 _pid,
+        address _account
+    ) public view returns (PoolBalances memory) {
+        IBaseRewardPool pool = IBaseRewardPool(_rewardPool);
+        uint256 staked = pool.balanceOf(_account);
+        uint256 earned = pool.earned(_account);
+
+        uint256 extraRewardsLength = pool.extraRewardsLength();
+        uint256[] memory extraRewardsEarned = new uint256[](extraRewardsLength);
+        for (uint256 i = 0; i < extraRewardsLength; i++) {
+            IBaseRewardPool extraRewardsPool = IBaseRewardPool(pool.extraRewards(i));
+            extraRewardsEarned[i] = extraRewardsPool.earned(_account);
+        }
+
+        return PoolBalances({ pid: _pid, staked: staked, earned: earned, extraRewardsEarned: extraRewardsEarned });
+    }
+
+    function getTokens(address[] memory _addresses) public view returns (Token[] memory) {
+        uint256 length = _addresses.length;
+        Token[] memory tokens = new Token[](length);
+
+        for (uint256 i = 0; i < length; i++) {
+            address addr = _addresses[i];
+            IERC20Detailed token = IERC20Detailed(addr);
+
+            uint8 decimals;
+            try token.decimals() {
+                decimals = token.decimals();
+            } catch {
+                decimals = 0;
+            }
+
+            tokens[i] = Token({ addr: addr, decimals: decimals, symbol: token.symbol(), name: token.name() });
+        }
+
+        return tokens;
+    }
+}
```

Comment:
This might be useful to show users the reward of triggering harvest on a UI, especially for cross chain:

```suggestion
function getEarmarkingReward(uint pool, address booster, address token) public returns(uint pending){
        uint start = IERC20(token).balanceOf(address(this));
        IBooster(booster).earmarkRewards(pool);
        pending = IERC20(token).balanceOf(address(this)) - start;
    }

    function getMultipleEarmarkingRewards(uint[] memory pools, address booster, address token) external returns(uint[] memory pendings){
        pendings = new uint[](pools.length);
        for(uint i =0 ; i < pools.length; i++){
            pendings[i] = getEarmarkingReward(pools[i], booster, token);
        }
    }
}
```